### ISSUE TITLE
Firebase Auth returns 500s

### ISSUE THREAD

**Issue by DeJayDev** [honojs/middleware#335]
Hi, when using the example for `@hono/firebase-auth`, a Hono server returns `500 Internal Server Error` instead of the expected proper error codes.

My reproduction case is the one provided in the `@hono/firebase-auth` README,

```ts
import { Hono } from 'hono'
import {
  VerifyFirebaseAuthConfig,
  VerifyFirebaseAuthEnv,
  verifyFirebaseAuth,
  getFirebaseToken,
} from '@hono/firebase-auth'

const config: VerifyFirebaseAuthConfig = {
    projectId: 'lessgames-73fa6',
}

const app = new Hono<{ Bindings: VerifyFirebaseAuthEnv }>()

app.use('*', verifyFirebaseAuth(config))
app.get('/hello', (c) => {
  const idToken = getFirebaseToken(c)
  return c.json(idToken)
})

export default app
```

Where I've also defined `PUBLIC_JWK_CACHE_KV` and `PUBLIC_JWK_CACHE_KEY` in my `wrangler.toml`. Though I have `node_compat` enabled as well, this issue exists regardless.

Running `wrangler dev src/index.ts` and hitting the  `/` endpoint with zero Authorization provides the following:

<details><summary>Wrangler Remote Error</summary>

<p>

```
Your worker has access to the following bindings:
- KV Namespaces:
  - PUBLIC_JWK_CACHE_KV: 042451a9970b4eadb0bb59fcba417223
- R2 Buckets:
  - MY_BUCKET: lessgames-preview
- Vars:
  - PUBLIC_JWK_CACHE_KEY: "jwk-key"
[wrangler:inf] Ready on http://localhost:8787
Total Upload: 147.25 KiB / gzip: 30.74 KiB
[wrangler:inf] GET / 500 Internal Server Error (1683ms)
Error
    at Array.<anonymous> (file:///<path>/.wrangler/tmp/dev-yqZ34O/index.js:2789:17)
    at Hono2.dispatch (file:///<path>/.wrangler/tmp/dev-yqZ34O/index.js:3937:38)
    at Hono2.fetch (file:///<path>/.wrangler/tmp/dev-yqZ34O/index.js:3777:19) {
  res: Response,
  status: 400,
  stack: Error
    at Array.<anonymous> (file:///<path>/.wrangler/tmp/dev-6ovetY/index.js:3777:19)
}

```
</p>
</details> 

---

| Package | Version |
|--------|--------
| `hono` | `3.11.12` | 
| `@hono/firebase-auth` | `1.3.3` | 
| `wrangler` | `3.22.1` |

**Comment by Code-Hex** [comment]
@yusukebe Could you check this one please.

Firebase Auth returns 400, but Hono responds as 500 in not found routing

**Comment by yusukebe** [comment]
Hi @DeJayDev , Thank you for raising the issue!
Hi @Code-Hex , Thank you for handling it!

This issue may be fixed with the PR #337. Try the latest version `v1.3.4` of `@hono/firebase-auth`!
